verbit/streaming_client.py

```diff
@@ -118,24 +135,20 @@ def start_stream(self,
             target=self._media_sender_worker,
             args=(media_generator, ),
             name='ws_media_sender')
+        self._stop_media_thread = False
         self._media_sender_thread.start()
 
         # return response generator
         return self._response_generator()
 
     def send_event(self, event: str, payload: dict = None):
+        if self._ws_client is None or not self._ws_client.connected:
+            raise RuntimeError('WebSocket client is disconnected!')
+        self._send_event(event, payload)
 
-        # use default payload if not provided
-        payload = payload or dict()
-
-        # prepare message dict
-        msg = dict(event=event, payload=payload)
-
-        # serialize as json
-        msg_json = json.dumps(msg)
-
-        # send to server
-        self._ws_client.send(msg_json)
+    def send_eos_event(self):
+        self._logger.debug(f'Sending EOS event')
```

Comment:
if you want this debug log to stay in the code, better move it to `send_event`
so all events are logged when debug is enabled